app/scripts/modules/cloudfoundry/src/loadBalancer/configure/CloudFoundryMapLoadBalancerModal.tsx

```diff
@@ -0,0 +1,216 @@
+import { AccountRegionClusterSelector, Routes } from 'cloudfoundry/presentation';
+
+export interface ICreateCloudFoundryMapLoadBalancerState {
+  accounts: IAccount[];
+  regions: string[];
+  application: Application;
+  selectedValues: ICloudFoundryLoadBalancerModalValues;
+  taskMonitor: TaskMonitor;
+}
+
+export interface ICloudFoundryLoadBalancerModalProps extends IModalComponentProps {
+  application: Application;
+}
+
+export interface ICloudFoundryLoadBalancerModalValues {
+  credentials: string;
+  region: string;
+  cluster: string;
+  routes: string[];
+}
+
+export class CloudFoundryMapLoadBalancerModal extends React.Component<
+  ICloudFoundryLoadBalancerModalProps,
+  ICreateCloudFoundryMapLoadBalancerState
+> {
+  public static defaultProps: Partial<ICloudFoundryLoadBalancerModalProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  private destroy$ = new Subject();
+  private formikRef = React.createRef<Formik<ICloudFoundryLoadBalancerModalValues>>();
+  private refreshUnsubscribe: () => void;
+  private $uibModalInstanceEmulation: IModalServiceInstance & { deferred?: IDeferred<any> };
+
+  constructor(props: ICloudFoundryLoadBalancerModalProps) {
+    super(props);
+
+    const deferred = $q.defer();
+    const promise = deferred.promise;
+    this.$uibModalInstanceEmulation = {
+      result: promise,
+      close: () => this.props.dismissModal(),
+      dismiss: () => this.props.dismissModal(),
+    } as IModalServiceInstance;
+    Object.assign(this.$uibModalInstanceEmulation, { deferred });
+    this.state = {
+      accounts: [],
+      regions: [],
+      application: props.application,
+      selectedValues: {
+        credentials: '',
+        region: '',
+        cluster: '',
+        routes: [''],
+      },
+      taskMonitor: new TaskMonitor({
+        application: props.application,
+        title: 'Mapping a route to your server group',
+        modalInstance: TaskMonitor.modalInstanceEmulation(() => this.props.dismissModal()),
+        onTaskComplete: () => this.props.application.serverGroups.refresh(),
+      }),
+    };
+    Observable.fromPromise(AccountService.listAccounts('cloudfoundry'))
+      .takeUntil(this.destroy$)
+      .subscribe((rawAccounts: IAccount[]) => this.setState({ accounts: rawAccounts }));
+  }
+
+  public static show(props: ILoadBalancerModalProps): Promise<void> {
+    const modalProps = { dialogClassName: 'wizard-modal modal-lg' };
+    return ReactModal.show(
+      CloudFoundryMapLoadBalancerModal,
+      {
+        application: props.app,
+        // className: 'create-pipeline-modal-overflow-visible',
+      },
+      modalProps,
+    );
+  }
+
+  public componentDidMount(): void {
+    Observable.fromPromise(AccountService.listAccounts('cloudfoundry'))
+      .takeUntil(this.destroy$)
+      .subscribe((accounts) => this.setState({ accounts }));
+  }
+
+  public componentWillUnmount(): void {
+    if (this.refreshUnsubscribe) {
+      this.refreshUnsubscribe();
+    }
```

Comment:
refreshUnsubscribe seems to be noop and can be removed

Reply:
Removed as well.